### PR TITLE
guardian.py

### LINKED ISSUE #8: Create SQLAlchemy Guardian Model Based on UI Mockup
As part of the development for the student registration system API, we need to create SQLAlchemy models that align with the UI mock up and requirements. These models will serve as the foundation for storing student registration data.
Guardian Model:

    id: Integer, primary key, unique identifier.
    student:   -> foreign to student model
    relationship: tuple, relationship of guardian.
    first_name: String, first name of guardian.
    last_name: String, last name of guardian.
    middle_name: String, middle name of guardian.

   

Model Validation:

    Ensure required fields are non-null and correctly validated.

Database Migration:

    Generate a migration to create the students table.

### PR DESCRIPTION
**Pull request by skadjei**
**Issue**

- Link:
- Summary:

**How**

**Screenshots**

**References**

**Checklist**

- [ ] Tests added
- [ ] Cleaned up commit history (rebase!)
- [ ] Documentation added
- [ ] Thought about variable and method names

### CODE REVIEW

File: app/models/guardian.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import uuid
+from sqlmodel import Field, SQLModel, Relationship
+from enum import Relationship
+
+
+class Guardian(SQLModel, table=True):
+    int: uuid.UUID = Field(default_factory=uuid.uuid4, primary_key=True)
```
Comment (Fourteen98) [inline app/models/guardian.py]: @mastermind-zed , use id int: Optional[int] = Field(default=None, primary_key)
you can import Optional from typing

File: app/models/guardian.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    first_name: str
+    last_name: str
+    middle_name: str
+    relationship: Relationship
+    student_id: int = Field(foriegn_key="student.id")
+
+    student = Relationship("Student", back_populates="guardians")
```
Comment (Fourteen98) [inline app/models/guardian.py]: Remove this line of code

File: app/models/guardian.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+from enum import Relationship
+
+
+class Guardian(SQLModel, table=True):
+    int: uuid.UUID = Field(default_factory=uuid.uuid4, primary_key=True)
+    first_name: str
+    last_name: str
+    middle_name: str
+    relationship: Relationship
+    student_id: int = Field(foriegn_key="student.id")
```
Comment (Fourteen98) [inline app/models/guardian.py]: you have a typo at the foreign_key, and the key is "students.id"

**Review by Fourteen98 — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @mastermind-zed  :wave:

## Highlights :1st_place_medal:

:heavy_check_mark: Following GitFlow
👎🏽 : Descriptive PR Title
👎🏽 : Descriptive PR Summary

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

File: app/models/guardian.py (hunk 2 above)
Comment (skadjei) [reply app/models/guardian.py]: line 11 to 14?

**Review by Fourteen98 — APPROVED** [review]
### **STATUS: APPROVED  ✔️** 
------
Hi @mastermind-zed   ,  

Your feature is complete! Great job 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍 
- No linter errors ✔️ 



image

Cheers and Happy coding! :clap: :clap: :clap:

**Pull request merged**